backend/src/backend/db/models.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime
+from ipaddress import IPv4Address
+from typing import List, Optional
+from uuid import UUID
+
+from sqlalchemy import DateTime, Enum, ForeignKey, UniqueConstraint, text
+from sqlalchemy.dialects.postgresql import ARRAY, CITEXT, INET
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    MappedAsDataclass,
+    mapped_column,
+    relationship,
+)
+from sqlalchemy.sql.schema import MetaData
+
+from backend.enums import StatusEnum
+
+
+class Base(MappedAsDataclass, DeclarativeBase):
+    # This map details the specific transformation of types between Python and
+    # PostgreSQL. This is only needed for the case where a specific PostgreSQL
+    # type has to be used.
+
+    type_annotation_map = {  # noqa: RUF012
+        str: CITEXT,  # transform Python str to PostgreSQL CITEXT
```

Comment:
I'm not convinced by the use of `citext` (that I didn't known).
First of all, mapping all `str` to it is probably a bad idea as it breaks the explicit python convention.

Using it on some columns ? Which ones? Why? I don't see which scenario (queries) would benefit from it. [documentation](https://www.postgresql.org/docs/current/citext.html) is not much convincing (see Tip and limitations)

Reply:
Okay, I was thinking that since the data we are going to be storing is from sources that we can't really control - the mirror list page, the country name and code from an external API, the `citext` data type would ensure queries are case-insenstive. A query on the backend requesting for data on `Nigeria` would still return entries for `nigeria` (assuming the admin who registered the mirror uses lowercase throughout).

Reply:
Should I revert this to native text characters regardless?

Reply:
I think it's better for now. So we don't have to think about it. We can still either normalize in core or alter DB should the need arise later

Reply:
Okay. I've completed all the requested changes.

Reply:
OK then:
- push them (I believe it's already done)
- _Resolve_ the conversations that you solved: if you changed something as requested for instance. If it was a question or if it's handled but differently than requested, let open for the reviewer to resolve.
- Re-request a review (from me)

Reply:
This is still using `CITEXT`… ???